packages/devtools_app/lib/src/profiler/cpu_profiler.dart

```diff
@@ -253,3 +267,73 @@ class CpuProfilerDisabled extends StatelessWidget {
     );
   }
 }
+
+
+/// DropdownButton that controls the value of
+/// [ProfilerScreenController.userTagFilter].
+class UserTagDropdown extends StatelessWidget {
+  const UserTagDropdown(this.controller);
+
+  final CpuProfilerController controller;
+
+  @override
+  Widget build(BuildContext context) {
+    const filterByTag = 'Filter by tag:';
+    // This needs to rebuild whenever there is new CPU profile data because the
+    // user tags will change with the data.
+    return ValueListenableBuilder<CpuProfileData>(
+      valueListenable: controller.dataNotifier,
+      builder: (context, cpuProfileData, _) {
+        return ValueListenableBuilder<String>(
+          valueListenable: controller.userTagFilter,
+          builder: (context, userTag, _) {
+            final userTags = controller.userTags ?? [];
+            final tooltip = userTags.isNotEmpty
+                ? 'Filter the CPU profile by the given UserTag'
+                : 'No UserTags found for this CPU profile';
+            return DevToolsTooltip(
+              tooltip: tooltip,
+              child: RoundedDropDownButton<String>(
+                isDense: true,
+                style: Theme.of(context).textTheme.bodyText2,
+                value: userTag,
+                items: [
+                  _buildMenuItem(
+                    display:
+                        '$filterByTag ${CpuProfilerController.userTagNone}',
+                    value: CpuProfilerController.userTagNone,
+                  ),
+                  // We don't want to show the 'Default' tag if it is the only
+                  // tag available. The 'none' tag above is equivalent in this
+                  // case.
+                  if (!(userTags.length == 1 &&
+                      userTags.first == UserTag.defaultTag.label))
+                    for (final tag in userTags)
+                      _buildMenuItem(
+                        display: '$filterByTag $tag',
+                        value: tag,
+                      ),
+                ],
+                onChanged: userTags.isNotEmpty ? _onUserTagChanged : null,
```

Comment:
shouldn't this be items.length > 1 instead of userTags.isNotEmpty?
I expect you don't really want this to be selectable if the only tag was the default tag.

Reply:
this is what this code is doing. `items` above will always have one item because we are showing a `none` option to negate any filters and show all samples. Then after that item, we will add the items from `userTags`, so if `userTags` is empty, `items` will have length 1 with just the `none` option. If `userTags` has length 1, then the two options we'd want to show are the one user tag and the none item.